engine/src/main/java/org/destinationsol/game/Hero.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.destinationsol.game;
+
+import com.badlogic.gdx.math.Vector2;
+import org.destinationsol.common.SolDescriptiveException;
+import org.destinationsol.game.input.Pilot;
+import org.destinationsol.game.item.Armor;
+import org.destinationsol.game.item.ItemContainer;
+import org.destinationsol.game.item.Shield;
+import org.destinationsol.game.item.SolItem;
+import org.destinationsol.game.item.TradeContainer;
+import org.destinationsol.game.ship.FarShip;
+import org.destinationsol.game.ship.ShipAbility;
+import org.destinationsol.game.ship.SolShip;
+import org.destinationsol.game.ship.hulls.Hull;
+
+/**
+ * A wrapper class for the Hero, that handles the normal and transcendent ships transparently.
+ */
+public class Hero {
+    private SolShip hero;
+    private StarPort.Transcendent transcendentHero;
+    private FarShip transcendentHeroShip;
+    private boolean isTranscendent;
+    private boolean isDead;
+
+    public Hero(SolShip hero) {
+        this.hero = hero;
+        if (hero == null) {
+            throw new SolDescriptiveException("Something tries to create hero, when there is no ship linked to hiim. Please report this..");
+        }
+        isTranscendent = false;
+    }
+
+    public void toTranscendent(StarPort.Transcendent transcendentHero) {
+        this.transcendentHero = transcendentHero;
+        transcendentHeroShip = transcendentHero.getShip();
+        isTranscendent = true;
+    }
+
+    public void toSolShip(SolShip hero) {
+        isDead = false;
+        this.hero = hero;
+        isTranscendent = false;
+    }
+
+    public boolean isTranscendent() {
+        return isTranscendent;
+    }
+
+    public boolean isNonTranscendent() {
+        return !isTranscendent;
+    }
+
+    public Pilot getPilot() {
+        return isTranscendent ? transcendentHeroShip.getPilot() : hero.getPilot();
+    }
+
+    public SolShip getHero() {
+        if (isTranscendent) {
+            throw new SolDescriptiveException("Something is trying to get a SolShip hero while the hero is Transcendent state.");
```

Comment:
Maybe briefly explain why that matters in a parenthesis at the end of the string "(this makes no sense because X)"

Reply:
I kinda see no reason. Imagine you have a bottle with blue ball, and you try to take the red out. This is about the same.

Reply:
Are you trying to say we're in the Matrix? :D

(Speaking as somebody who still isn't totally sure what transcendent state even is it just stuck out to me)

Reply:
This is basically the same thing. Hero is just some kind of bottle, that may contain either SolShip hero or Transcendent hero. This exception says that you are trying to get the SolShip hero while there is Transcendent hero (and thus, no solhip hero) (previously, before the Hero, there was just null-checking in EVERY SINGLE PLACE that used `game.getHero()` (well, actually `g.getHero()`, unless `g` was gun, in which case it was `ga`))